src/game/lara1gun.c

```diff
@@ -164,27 +164,22 @@ void AnimateShotgun()
             }
         }
     } else {
-        if (ani == AF_SG_AIM && Input.action) {
-            ani++;
-        } else if (ani > AF_SG_AIM && ani < AF_SG_DRAW) {
-            ani++;
-            if (ani == AF_SG_DRAW) {
-                ani = Input.action ? AF_SG_RECOIL : AF_SG_UNAIM;
-            }
-        } else {
-            if (ani == AF_SG_RECOIL) {
+        if (T1MConfig.enable_shotgun_target_fix && Input.action
+            && !Lara.target) {
+            if (ani >= AF_SG_AIM && ani < AF_SG_DRAW) {
+                ani++;
+                if (ani == AF_SG_DRAW) {
+                    ani = AF_SG_RECOIL;
+                }
+            } else if (ani == AF_SG_RECOIL) {
                 if (Input.action) {
                     FireShotgun();
                     ani++;
-                } else {
-                    ani = AF_SG_UNAIM;
                 }
             } else if (ani > AF_SG_RECOIL && ani < AF_SG_UNDRAW) {
                 ani++;
-                if (ani == AF_SG_RECOIL + 12 + 1) {
-                    ani = AF_SG_AIM;
-                } else if (ani == AF_SG_UNDRAW) {
-                    ani = AF_SG_UNAIM;
+                if (ani == AF_SG_UNDRAW) {
+                    ani = AF_SG_RECOIL;
```

Comment:
In this case, OG goes to `AF_SG_UNAIM`. Why the change?

Reply:
So I changed the fixed to be more like the `if (Lara.left_arm.lock)` part of the code, and less like what is now `else if (!T1MConfig.enable_shotgun_target_fix)`. I definitely do notice what you are saying now about the unhandled case where you stop holding action and Lara's animation gets stuck at whatever point you were last holding action at. I'll do some more tweaking.

Originally I thought they devs had meant to implement a third case where Lara is meant to fire quicker with subsequent shots after the first shot at a locked target. But, instead it was firing quicker when there is no locked target. However, I decided to go with the original behavior which is a full animation for every shot, and fix the fast shooting at no target and the shooting when a locked target is out of her sight.

Reply:
Now I understand it better :) I'll give all of this a re-read tomorrow and really try to get how it works, my head is done for today. Of course, feel free to push any changes.